src/openloco/windows/music_selection.cpp

```diff
@@ -0,0 +1,236 @@
+        make_widget({ 0, 0 }, { 360, 238 }, widget_type::frame, 0),
+        make_widget({ 1, 1 }, { 358, 13 }, widget_type::caption_25, 0, string_ids::music_selection_title),
+        make_widget({ 345, 2 }, { 13, 13 }, widget_type::wt_9, 0, image_ids::close_button, string_ids::tooltip_close_window),
+        make_widget({ 0, 15 }, { 360, 223 }, widget_type::panel, 1),
+        make_widget({ 4, 19 }, { 352, 218 }, widget_type::scrollview, 1, scrollbars::vertical, string_ids::music_selection_tooltip),
+        widget_end(),
+    };
+
+    static window_event_list _events;
+
+    static void draw(ui::window* window, gfx::drawpixelinfo_t* dpi);
+    static void draw_scroll(ui::window* window, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex);
+    static void get_scroll_size(ui::window* window, uint32_t scrollIndex, uint16_t* scrollWidth, uint16_t* scrollHeight);
+    static void on_mouse_up(ui::window* window, widget_index widgetIndex);
+    static void on_scroll_mouse_down(ui::window* window, int16_t x, int16_t y, uint8_t scroll_index);
+    static void on_scroll_mouse_over(ui::window* window, int16_t x, int16_t y, uint8_t scroll_index);
+    static void on_update(window* window);
+    static void tooltip(ui::window* window, widget_index widgetIndex);
+
+    static void init_events()
+    {
+        _events.draw = draw;
+        _events.draw_scroll = draw_scroll;
+        _events.get_scroll_size = get_scroll_size;
+        _events.on_mouse_up = on_mouse_up;
+        _events.on_update = on_update;
+        _events.scroll_mouse_down = on_scroll_mouse_down;
+        _events.scroll_mouse_over = on_scroll_mouse_over;
+        _events.tooltip = tooltip;
+    }
+
+    // 0x004C1602
+    window* open()
+    {
+        window* window = WindowManager::bringToFront(WindowType::musicSelection, 0);
+        if (window != nullptr)
+            return window;
+
+        window = WindowManager::createWindow(
+            WindowType::musicSelection,
+            { ww, wh },
+            0,
+            &_events);
+
+        // TODO: only needs to be called once.
+        init_events();
+
+        window->widgets = _widgets;
+        window->enabled_widgets = 1 << widx::close;
+        window->init_scroll_widgets();
+
+        auto interface = objectmgr::get<interface_skin_object>();
+        window->colours[0] = interface->colour_0B;
+        window->colours[1] = interface->colour_10;
+
+        window->row_count = numTracks;
+        window->row_hover = -1;
+
+        return window;
+    }
+
+    // 0x004C165D
+    static void draw(ui::window* window, gfx::drawpixelinfo_t* dpi)
+    {
+        // Draw widgets.
+        window->draw(dpi);
+    }
+
+    // 0x004C1663
+    static void draw_scroll(ui::window* window, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex)
+    {
+        auto shade = colour::get_shade(window->colours[1], 4);
+        gfx::clear_single(*dpi, shade);
+
+        static const string_id music_ids_to_string_id[] = {
```

Comment:
This is the second occurence of this list. Add string_ids to MusicInfo (`audio.cpp`, integrated version of 0x004FE910) and update options screen to get it from `audio.cpp` as well

Reply:
Should probably merge in the similar list from the 'About music' window while we're at it.

https://github.com/OpenLoco/OpenLoco/blob/632f4a932cdd051a4c531d1dda1893b454f0650a/src/openloco/windows/about_music.cpp#L109-L140